src/main/scala/util/RunUtils.scala

```diff
@@ -221,6 +221,13 @@ object IRTransform {
     Logger.debug(
       s"[!] Removed ${before - ctx.program.procedures.size} functions (${ctx.program.procedures.size} remaining)"
     )
+    val dupProcNames = (ctx.program.procedures.groupBy(_.name).filter((n,p) => p.size > 1)).toList.flatMap(_._2)
+
+    var dupCounter = 0 
+    for (p <- dupProcNames) {
+      dupCounter += 1
+      p.name = p.name + "$" + p.address.map(_.toString).getOrElse(dupCounter.toString)
+    }
```

Comment:
This is good to have but may potentially cause problems elsewhere because there are some things (like indirect call resolution) that rely on the names of procedures in a somewhat fragile way. That is something that I want to fix in the indirect call resolution though.

Reply:
Yes its not correct to rely on names it should go through symbol table address -> basil procedure address, we should probably be clear about what the correct unique procedure identifier is in the IR though.